fictadvisor-api/src/v2/utils/parser/RozParser.ts

```diff
@@ -272,57 +272,48 @@ export class RozParser implements Parser {
       ],
     });
 
-    const weeksCountEveryFortnight = await this.getWeeksCount(startOfEvent, Period.EVERY_FORTNIGHT);
-    const weeksCountEveryWeek = await this.getWeeksCount(startOfEvent, Period.EVERY_WEEK);
-
     if (!event) {
       await this.eventRepository.create({
         name: pair.name,
         startTime: startOfEvent,
         endTime: endOfEvent,
         period: Period.EVERY_FORTNIGHT,
+        eventsAmount: await this.getWeeksCount(startOfEvent, Period.EVERY_FORTNIGHT),
         groupId: groupId,
         lessons: {
           create: {
             disciplineTypeId: disciplineType.id,
           },
         },
-        eventInfo: {
-          createMany: {
-            data: this.getIndexesForEventInfo(0,  weeksCountEveryFortnight - 1),
-          },
-        },
       });
     } else if (
       event.startTime.getTime() === startOfEvent.getTime() - WEEK &&
       event.period === Period.EVERY_FORTNIGHT
     ) {
       await this.eventRepository.updateById(event.id, {
         period: Period.EVERY_WEEK,
-        endTime: endOfEvent,
-        eventInfo: {
-          createMany: {
-            data: this.getIndexesForEventInfo(weeksCountEveryFortnight, weeksCountEveryWeek),
-          },
-        },
+        eventsAmount: await this.getWeeksCount(startOfEvent, Period.EVERY_WEEK),
       });
     }
   }
 
-  getIndexesForEventInfo (startIndex: number, endIndex: number) {
-    return Array.from({
-      length: endIndex - startIndex + 1,
-    }, (_, i) => ({
-      number: i + startIndex,
-    }));
+  async getWeeksCount (startOfEvent: Date, period: Period) {
```

Comment:
I think there is just so many not checked mathematical logic, that you write VERY OVERHELMING code that is not necessary.

NO MATTER is there the first or the second week. You can count by yourself amount of fortnight laboratory works and you will see that there are 9 pair (no need to check first week or the second) and 18 lectures for example

You can simplify it just by writing:
```ts
const { startDate, endDate } = await this.dateService.getCurrentSemester();
const lastWeek = DateUtils.getDifference(startDate, endDate, WEEK);
return lastWeek / weeksForEvent[period];
```

thats all